client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -551,11 +551,12 @@ public void close() throws SegmentSealedException {
         }
     }
 
+
     /**
      * @see SegmentOutputStream#flush()
      */
     @Override
-    public void flush() throws SegmentSealedException {
+    public CompletableFuture<Boolean> flushAsync() throws SegmentSealedException {
```

Comment:
I think we are not replacing the existing `flush` method but just adding a new `flushAsync` API. This change makes it somewhat confusing to review because the `flush` method has been moved down and replaced with flushAsync. It may be better to maintain the existing method as is and just add the new one....

Reply:
That makes sense, I updated it and moved the flushAsync beneath the flush method.